solvers/amigo.py

```diff
@@ -190,7 +190,7 @@ def run(self, callback):
                 self.f_inner, inner_var, outer_var, self.step_size,
                 sampler=inner_sampler, n_steps=self.n_inner_steps,
             )
-        while callback((inner_var, outer_var)):
+        while callback(dict(inner_var=inner_var, outer_var=outer_var)):
```

Comment:
the new api uses `get_result` to access the result in the callback

```suggestion
        self.inner_var, self.outer_var = inner_var, outer_var
        while callback():
```